src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -337,7 +348,15 @@ pg_get_tableschemadef_string(Oid tableRelationId, bool includeSequenceDefaults)
 	{
 		Form_pg_attribute attributeForm = tupleDescriptor->attrs[attributeIndex];
 
-		if (!attributeForm->attisdropped && attributeForm->attinhcount == 0)
```

Comment:
No, we don't. Currently what happens is that we allow creating those tables, but, the inheritance hierarchy is not created on the workers. Do you prefer erroring out for such cases?